src/integrationTest/cloudformation/templateRegistryManager.test.ts

```diff
@@ -19,118 +19,116 @@ import { getTestWorkspaceFolder } from '../integrationTestsUtilities'
-                await rmrf(file)
-            }
-        }
         manager.dispose()
+        await rmrf(testDir)
     })
 
     it('adds initial template files with yaml and yml extensions at various nesting levels', async () => {
+        await createTestYamlTemplates(testDir, testDirNested)
+
         await manager.addTemplateGlob('**/test.{yaml,yml}')
-        await new Promise(resolve => setTimeout(resolve, 500))
-        assert.strictEqual(registry.registeredTemplates.length, 2)
+
+        assert.strictEqual(await registryHasTargetNumberOfFiles(registry, 2), true)
     })
 
     it('adds dynamically-added template files with yaml and yml extensions at various nesting levels', async () => {
         await manager.addTemplateGlob('**/test.{yaml,yml}')
-        await new Promise(resolve => setTimeout(resolve, 500))
-        assert.strictEqual(registry.registeredTemplates.length, 2)
-
-        const path1 = path.join(testDir, 'test.yml')
-        const path2 = path.join(testDir, 'nested', 'test.yaml')
-        additionalFiles.push(path1)
-        additionalFiles.push(path2)
 
-        const filename = vscode.Uri.file(path1)
-        await strToYamlFile(makeSampleSamTemplateYaml(false), filename.fsPath)
-
-        const filename2 = vscode.Uri.file(path2)
-        await strToYamlFile(makeSampleSamTemplateYaml(false), filename2.fsPath)
+        await createTestYamlTemplates(testDir, testDirNested)
 
-        await new Promise(resolve => setTimeout(resolve, 500))
-        assert.strictEqual(registry.registeredTemplates.length, 4)
+        assert.strictEqual(await registryHasTargetNumberOfFiles(registry, 2), true)
     })
 
     it('can handle changed files', async () => {
-        let objectsFound = 0
+        const filepath = path.join(testDir, 'changeMe.yml')
 
-        const path1 = path.join(testDir, 'changeMe.yml')
-        additionalFiles.push(path1)
-
-        const filename = vscode.Uri.file(path1)
+        const filename = vscode.Uri.file(filepath)
         await strToYamlFile(makeSampleSamTemplateYaml(false), filename.fsPath)
 
         await manager.addTemplateGlob('**/changeMe.yml')
 
-        await new Promise(resolve => setTimeout(resolve, 500))
-        assert.strictEqual(registry.registeredTemplates.length, 1)
+        assert.strictEqual(await registryHasTargetNumberOfFiles(registry, 1), true)
 
-        const initialObj = registry.getRegisteredTemplate(path1)
+        const initialObj = registry.getRegisteredTemplate(filepath)
         assert.ok(initialObj)
         if (initialObj) {
-            objectsFound++
             assert.strictEqual(Object.keys(initialObj.template).includes('Globals'), false)
         }
 
         await strToYamlFile(makeSampleSamTemplateYaml(true), filename.fsPath)
 
         await new Promise(resolve => setTimeout(resolve, 500))
-        assert.strictEqual(registry.registeredTemplates.length, 1)
+        assert.strictEqual(await registryHasTargetNumberOfFiles(registry, 1), true)
```

Comment:
Because you're editing the same file, this check isn't valuable.
You'll want to loop until you get a registered template for filepath that contains a Globals section.